main/inc/lib/tracking.lib.php

```diff
@@ -3347,6 +3347,104 @@ public static function count_student_exercise_attempts(
         return $count_attempts;
     }
 
+    /**
+     * It gets the last finalization date of learnpaths in a course.
+     *
+     * @param int  $userId
+     * @param int  $courseId
+     * @param int  $sessionId
+     * @param bool $convertDate
+     *
+     * @return false|mixed|string
+     */
+    public static function getCourseLpFinalizationDate(
+        $userId,
+        $courseId,
+        $sessionId,
+        $convertDate = true
```

Comment:
This should use parameter types in the function declaration rather than in PHPDoc. However, don't define the return type as multiple return types are only authorized since PHP 8.0